scripts/dts/edtlib.py

```diff
@@ -1144,6 +1224,107 @@ def _check_undeclared_props(self):
                      .format(prop_name, self._node.path, self.edt.dts_path,
                              self.binding_path))
 
+    def _init_gpio_ranges(self):
+        # Initialize self.pinctrl_gpio_ranges
+
+        if not hasattr(self, 'pinctrl_gpio_ranges'):
+            self.pinctrl_gpio_ranges = []
+
+        if 'gpio-ranges' not in self.props:
+            return
+
+        for prop_value in self.props['gpio-ranges'].val:
+            entry = ControllerAndData()
+            entry.node = prop_value.node
+            entry.name = prop_value.name
+            entry.controller = self
+            entry.data = prop_value.data
+            # Add to pin controller
+            if not hasattr(prop_value.controller, 'pinctrl_gpio_ranges'):
+                prop_value.controller.pinctrl_gpio_ranges = []
+            prop_value.controller.pinctrl_gpio_ranges.append(entry)
+
+    def _init_clock_outputs(self):
```

Comment:
Quick look, but this function looks pretty long and hard to follow.

Could you try splitting it up into some smaller functions, so that each functions is like 30 lines maximum? Maybe some common stuff could be factored out at the same time.

Reply:
Split. Did not reach 30 lines. Where do get this limit from?

Reply:
Just some number I pulled out that feels like the right ballpark. Short and focused functions with few parameters are considered a good practice at least. See e.g. the [Linux kernel coding style](https://www.kernel.org/doc/html/v4.10/process/coding-style.html#functions).

Alright to have it a bit longer if it's just a big switch or whatever with only one or two levels of indent, because then it's more like many small functions.

Good idea to go through [PEP 8](https://www.python.org/dev/peps/pep-0008/) once too. Check [this section](https://www.python.org/dev/peps/pep-0008/#indentation) for how to format continuation lines (e.g. for function parameters).

Strive for having few parameters too, and for all combinations of parameters to make sense (e.g., no dependencies between parameters, if it can be avoided).

Might be a bit busy next week, but will look more at the logic.

Reply:
Not saying I'm perfect re. that stuff. Think it might be possible to shorten some of this stuff though, and have some simple helpers with just a few parameters.